website/util/metrics.py

```diff
@@ -1,14 +1,119 @@
 # -*- coding: utf-8 -*-
-
+from enum import Enum
 
 def get_entry_point(user):
     """
     Given the user system_tags, return the user entry point (osf, osf4m, prereg, institution)
     In case of multiple entry_points existing in the system_tags, return only the first one.
     """
-    entry_points = ['osf4m', 'prereg_challenge_campaign', 'institution_campaign', 'osf_preprints']
+    entry_points = [
+        CampaignSourceTags.Osf4m.value,
+        CampaignSourceTags.PreregChallenge.value,
+        'institution_campaign',
```

Comment:
One of these things is not like the others.

Are the plans to normalize this one at a later date, i.e. when we make general architecture improvements to OSF4I? If so, please document this need with a follow-up ticket or similar.

Reply:
The `get_entry_point()` hook is used only for keen analytics. And we plan on doing away with keen soon. The `institution_campaign` entry point itself is weird, as it doesn't seem to be used anywhere, and there is not a related system tag in the prod db. So I just leave it as is. I am not sure what the plan for OSF4I is.